datafiles/manager.py

```diff
@@ -33,31 +33,26 @@ def all(self) -> Iterator[HasDatafile]:
         for filename in iglob(splatted):
             log.debug(f'Found matching path: {filename}')
             results = parse(pattern, filename)
-            fields = dataclasses.fields(self.model)
-            args = list(results.named.values())
-            args += [Missing] * (len(fields) - len(args))
-            yield self.model(*args)
+            yield self.get(*results.named.values())
 
-    def get_or_none(self, *args, **kwargs) -> Optional[HasDatafile]:
-        original_manual = self.model.Meta.datafile_manual
-
-        self.model.Meta.datafile_manual = True
-        instance = self.model(*args, **kwargs)
-        self.model.Meta.datafile_manual = original_manual
+    def get(self, *args, **kwargs) -> HasDatafile:
```

Comment:
I hesitated between `get` and `get_or_raise`, but found that the first one made perfect sense. What do you think?

Reply:
Yeah, it seems reasonable to me for `get()` to raise an exception if the object does not exist.